fvm/blocks.go

```diff
@@ -1,9 +1,68 @@
 package fvm
 
-import "github.com/onflow/flow-go/model/flow"
+import (
+	"errors"
+	"fmt"
+	"github.com/onflow/flow-go/model/flow"
+	"github.com/onflow/flow-go/storage"
+)
 
 type Blocks interface {
-	// ByHeight returns the block at the given height. It is only available
-	// for finalized blocks.
-	ByHeight(height uint64) (*flow.Block, error)
+	// ByHeight returns the block at the given height in chain engine at header,
+	// or finalized if header is nil
```

Comment:
```suggestion
	// ByHeight returns the block at the given height in the chain ending in `header` (or finalized
	// if `header` is nil). This enables querying un-finalized blocks by height with respect to the
	// chain defined by the block we are executing.
```

Could add some more about why we need a reference block here